### PR TITLE
Add coldfusion support

### PR DESCRIPTION
**Pull request by finalcut**
used the instructions at https://github.com/Glavin001/atom-beautify/blob/master/docs/add-languages-and-beautifiers.md

also updated the prettydiff .coffee file as per the instructions at https://github.com/Glavin001/atom-beautify/issues/354#issuecomment-165817084

I *think* I got everything..  I don't know how or if I should have done something to get prettydiff 1.6.1 which is what adds CFML support to prettydiff itself.  https://github.com/prettydiff/prettydiff/issues/158

I also don't really know how to test this all out in Atom - I have no idea how to add a custom plugin to atom.  Hopefully this will, at least, be enough to get the ball rolling.

### CODE REVIEW

**Comment by finalcut** [comment]
I just updated package.json to point to js-beautify 1.6.1  hopefully that fixes the Travic CI problem.

All of your pull requests seem to have a problem with AppVeyor so I'm not sure what to do with that.

File: src/languages/coldfusion.coffee [hunk 1]
```diff
@@ -0,0 +1,25 @@
+module.exports = {
+
+  name: "CFML"
```
Comment (Glavin001) [inline src/languages/coldfusion.coffee]: What about changing this to `ColdFusion`? I'd like the `name`s for the languages to be full, and it also helps with documentation that is automatically generated.

Of course, if you change this name, as you know, remember to change in the `beautifiers/prettydiff.coffee`

**Comment by Glavin001** [comment]
Quick comments.

> All of your pull requests seem to have a problem with AppVeyor so I'm not sure what to do with that.

I will likely ignore AppVeyor. That is only for Windows support and always is causing more headache than it's worth.

> I just updated package.json to point to js-beautify 1.6.1 hopefully that fixes the Travic CI problem.

Travis CI is still failing, I see. Hopefully you can figure out how to get it working.
Thanks for adding the tests :+1:.

Also please generate the documentation in a separate commit. Run:

```bash
npm run docs
```

And commit the changes in the documentation.

Looking good so far! Thanks!

**Comment by finalcut** [comment]
I updated js-beautify and not prettydiff.. doh.

I'll fix that and it "should" fix the travis ci - I generated the expected with prettydiff directly.

**Comment by finalcut** [comment]
I don't really see how to run these tests locally.  I am not sure where atom-beautifier is on my machine in relationship to Atom; so I can't replace your local copy with mine to run the tests so I end up having to continue make travis run the tests which is pretty darn slow.

The travis ci output is pretty non-parsable by me - I can't tell what it is saying is wrong with the output.  I am basically just guessing on the various options to use in the `coldfusion.coffee` file and the `.jsbeautifyrc` file (I'm just copying the settings from the `html` one as I have no real idea what to put in these files and, what the various settings will actually do to the behavior of `prettydiff`.

Hopefully someone else can fix/finalize these things if the current build fails.

**Comment by Glavin001** [comment]
> I am not sure where atom-beautifier is on my machine in relationship to Atom; so I can't replace your local copy with mine to run the tests so I end up having to continue make travis run the tests which is pretty darn slow.

1. Git clone the Atom Beautify repo
2. Uninstall Atom Beautify, `apm uninstall atom-beautify`
3. In the Atom Beautify directory that you cloned, run `apm link`. This will link up your development copy of Atom Beautify with Atom.
4. Restart Atom and open your Atom Beautify clone directory
5. For running tests, Run command `Window: Run Package Spec`

> The travis ci output is pretty non-parsable by me

Look for the part that looks like:

```
--- /Users/travis/build/Glavin001/atom-beautify/examples/nested-jsbeautifyrc/coldfusion/expected/test.cfc	beautified
+++ /Users/travis/build/Glavin001/atom-beautify/examples/nested-jsbeautifyrc/coldfusion/expected/test.cfc	expected
@@ -1,8 +1,8 @@
 <cfcomponent␣output="no">⏎
-␣␣<cffunction␣name="init"␣output="false"␣returntype="any">⏎
-␣␣␣␣<cfscript>⏎
-␣␣␣␣␣␣variables.instance␣=␣structNew();⏎
-␣␣␣␣␣␣return␣this;⏎
-␣␣␣␣</cfscript>⏎
-␣␣</cffunction>⏎
+␣␣␣���<cffunction␣name="init"␣output="false"␣returntype="any">⏎
+␣␣␣␣␣␣␣␣<cfscript>⏎
+␣␣␣␣␣␣␣␣␣␣␣␣variables.instance␣=␣structNew();⏎
+␣␣␣␣␣␣␣␣␣␣␣␣return␣this;⏎
+␣␣␣␣␣␣␣␣</cfscript>⏎
+␣␣␣␣</cffunction>⏎
 </cfcomponent>
\ No newline at end of file
```

The lines that are prefixed with:
- `-` lines that the current beautification has that the expected (test) does not
- `+` lines that the expected (test) has but the current beautification does not

Goal would be to remove any `-` lines and ensure the `+` line exist when beautifying, as that is what the test expects the result to look like.

Sometime I'd like to add more to the contribution guidelines to help with this.

Hope this helps for now. Thanks!

**Comment by finalcut** [comment]
thanks for the extra info. I actually knew what part of the travis file was applicable I just didn't know what the <?> things were.. Nor do I know why in my expected result there are four spaces for a tab but in the beautification you do there are only 2.  

I tried to run this all locally but none of the tests execute properly.  For pretty much every beautification type test I get this result (image becuase I can't copy paste it for some reason):

image

**Comment by finalcut** [comment]
well travis failed for strange reasons that don't seem to have anything to do with my changes/test.

**Comment by Glavin001** [comment]
@prettydiff was the `MODULE_NOT_FOUND` from Pretty Diff? Those languages that are failing would be handled by Pretty Diff.

**Comment by prettydiff** [comment]
It could be related to https://github.com/prettydiff/prettydiff/issues/237  If so then just update prettydiff with

`npm install prettydiff@latest`

Currently, the latest version is 1.16.5.  Version 1.16.1 was missing its `main` file described in this bug: https://github.com/npm/npm/issues/5082

**Comment by Glavin001** [comment]
Please merge the latest, which should force 1.16.5 or higher for Pretty Diff. Thanks!

**Comment by prettydiff** [comment]
Sorry, I just realized the problem was in 1.16.5 also.  I bumped to 1.16.6 and verified the problem is resolved.  I getting killed between this NPM problem and this V8 issue: https://bugs.chromium.org/p/v8/issues/detail?id=4635

**Comment by Glavin001** [comment]
Please resolve merge conflicts and let the tests run again. Should be good to merge if they all pass after that. Thanks.

**Comment by finalcut** [comment]
well all sorts of other tests fail now.  - the parser (prettydiff) seems to be re-arranging attributes in a way the other tests didn't expect (such as the doctype tag in the html test, or the apex tag in the visualforce test)  

There also appears to be some spacing/tab issues (in the xml test)

**Comment by Glavin001** [comment]
No problem. I'll fix those unrelated issues on my own. Just looked over the changes and looks good. Would you be able to add `coldfusion` to the list of keywords in the `package.json`, that way I think the Atom package searches should work better if someone searches `coldfusion beautifier` or something similar. After that I think this is good to merge. Thanks!!

**Comment by finalcut** [comment]
Just an FYI I have addressed the description and keyword points you raised.  So this is finished and ready for you to merge.

**Comment by Glavin001** [comment]
Published to v0.28.23

**Pull request merged**